core/domain/fs_domain.py

```diff
@@ -135,16 +136,20 @@ class GeneralFileSystem(python_utils.OBJECT):
         entity_id: str. The ID of the corresponding entity.
     """
 
-    def __init__(self, entity_name, entity_id):
+    def __init__(
+            self, entity_name, entity_id):
         """Constructs a GeneralFileSystem object.
 
         Args:
             entity_name: str. The name of the entity
                 (eg: exploration, topic etc).
             entity_id: str. The ID of the corresponding entity.
+            user_id: str. The ID of the user on behalf of whom the file system
```

Comment:
This arg seems to be missing from the function signature.

Reply:
I will address all changes at once when you have done a complete pass. Till then, I will fix backend tests and perform end to end testing.

Reply:
Per our conversation: please

- remove the GAE datastore-backed system and the user ID
- add a full set of tests
- use OS-independent ways to construct paths

I'll take a full pass through the PR once these are done and provide a full list of comments, and after that, hopefully it will be mergeable!